akka-javasdk/src/main/java/akka/javasdk/CommandException.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2021-2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk;
+
+import akka.javasdk.eventsourcedentity.EventSourcedEntity;
+import akka.javasdk.keyvalueentity.KeyValueEntity;
+import akka.javasdk.workflow.Workflow;
+
+/**
+ * An exception that can be thrown by user code to signal domain validation errors or business rule violations.
+ * 
+ * <p>This exception is designed to be used in command handlers of {@link KeyValueEntity}, {@link EventSourcedEntity}, 
+ * or {@link Workflow} components when the incoming command doesn't fulfill the requirements or the current state 
+ * doesn't allow the command to be handled.
```

Comment:
A couple of question:

It's always it's usage tied to componentClient? If so I would add it here. If not I would add another example like: 
```
try {   return componentClient            .forEventSourcedEntity(counterId)            .method(CounterEntity::increase)            .invoke(value); } catch (CommandException e) {   // Handle the command exception, e. g., return a bad request response }
```

What would happen if used outside an entity (view or consumer). I guess nothing special but I would like to see it written down.